test/account-manager.js

```diff
@@ -0,0 +1,362 @@
+'use strict'
+
+const path = require('path')
+const fs = require('fs-extra')
+const chai = require('chai')
+const expect = chai.expect
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+chai.use(sinonChai)
+chai.should()
+
+const rdf = require('rdflib')
+const LDP = require('../lib/ldp')
+const SolidHost = require('../lib/models/solid-host')
+const AccountManager = require('../lib/models/account-manager')
+const UserAccount = require('../lib/models/user-account')
+const WebIdTlsCertificate = require('../lib/models/webid-tls-certificate')
+
+const testAccountsDir = path.join(__dirname, 'resources', 'accounts')
+
+var host
+
+beforeEach(() => {
+  host = SolidHost.from({ serverUri: 'https://example.com' })
+})
+
+describe('AccountManager', () => {
+  describe('from()', () => {
+    it('should init with passed in options', () => {
+      let config = {
+        host,
+        authMethod: 'tls',
+        multiUser: true,
+        store: {},
+        emailService: {}
+      }
+
+      let mgr = AccountManager.from(config)
+      expect(mgr.host).to.equal(config.host)
+      expect(mgr.authMethod).to.equal(config.authMethod)
+      expect(mgr.multiUser).to.equal(config.multiUser)
+      expect(mgr.store).to.equal(config.store)
+      expect(mgr.emailService).to.equal(config.emailService)
+    })
+  })
+
+  describe('accountUriFor', () => {
+    it('should compose account uri for an account in multi user mode', () => {
+      let options = {
+        multiUser: true,
+        host: SolidHost.from({ serverUri: 'https://localhost' })
+      }
+      let mgr = AccountManager.from(options)
+
+      let webId = mgr.accountUriFor('alice')
+      expect(webId).to.equal('https://alice.localhost')
+    })
+
+    it('should compose account uri for an account in single user mode', () => {
```

Comment:
I think the idea here is that it doesn't matter (for the purposes of the account uri method) what the username is, in single-user mode.